src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -104,7 +120,15 @@ void GcsPlacementGroupManager::OnPlacementGroupCreationFailed(
                    << ", try again.";
   // We will attempt to schedule this placement_group once an eligible node is
   // registered.
-  pending_placement_groups_.emplace_back(std::move(placement_group));
+  if (placement_group->GetState() == rpc::PlacementGroupTableData::RESCHEDULING) {
+    // NOTE: If a node is dead, the placement group scheduler should try to recover the
+    // group by rescheduling the bundles of the dead node. This should have higher
+    // priority than trying to place other placement groups, so we put it at the head of
+    // the queue.
+    pending_placement_groups_.emplace_front(std::move(placement_group));
```

Comment:
Good idea.

Reply:
Actually, I have one concern about this rescheduling priority. If we have a placement group bundle that requires a unique resource (for example gpu resource when there’s only one gpu node), this can postpone creating all other placement group until a node with the resources is added. Is there any good solution to resolve this issue?

Reply:
Actually, I have one concern about this rescheduling priority. If we have a placement group bundle that requires a unique resource (for example gpu resource when there’s only one gpu node), this can postpone creating all other placement group until a node with the resources is added. Is there any good solution to resolve this issue?

Reply:
`ResourceSet::IsSubset` increases the error code to return whether there is no such resource or the number of resources is insufficient. If all nodes do not have the required resources, the scheduling priority will be lowered(add a new PlacementGroupState `WAITING`), and the scheduling will be triggered when a new node is added.

Reply:
This policy sounds good to me. 

cc @ericl